### PR TITLE
Fix k8s modules and deployment tests

### PR DESCRIPTION
**Pull request by AlexBMes**
The k8s modules haven't been updated in a couple of years, so deployment tests have failed, as some of the resources have changed to needing `_v1` versions. Have updated all resource to be `_v1` resources, as they use the `v1` API, but resources without `_v1` seem to use `v1beta1` APIs now.

The changes in this PR have been tested in a minikube cluster.
## Breaking Changes
* Minimum kubernetes version is v1.23
* Update the minimum kubernetes provider version to `2.14.0` and max provider version `<3.0.0`
* Update all kubernetes resources to `v1` versions, apart from `kubernetes_horizontal_pod_autoscaler`  which is upgraded to v2 to maintain use of `metrics` configuration
* `ingress` now has `spec.default_backend` instead of `spec.backend`
* `spec.default_backend` and `spec.rule.http.path.*.backend` have a different format, allowing either `service` or `resource` blocks ( see KUBERNETES.md for example)
* `metric` block no longer available in `kubernetes_horizontal_pod_autoscaler_v1`
## Fixes
* Use different terraform workspaces for gcp and k8s deployment tests to avoid `terraform apply` destroying resources from other deployment (i.e. attempting to destroy app engine app)
* Add metadata configuration for cronjob job
* Add service to app_3 deployment test
## Changes
* Change logic behind port configuration in `kuberenetes_services_v1`
* New `container_resource` metric in k8s_pod_autoscaler

### CODE REVIEW

File: k8s_pod_autoscaler.tf [hunk 1]
```diff
@@ -20,113 +20,5 @@ resource "kubernetes_horizontal_pod_autoscaler" "self" {
         name        = lookup(scale_target_ref.value, "name", null)
       }
     }
-    dynamic "metric" {
```
Comment (gbmeuk) [inline k8s_pod_autoscaler.tf:23]: Is there no replacement for this?

Comment (AlexBMes) [reply k8s_pod_autoscaler.tf:23]: No, based on the [documetation ](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/horizontal_pod_autoscaler) for `kuberentes_horizontal_pod_autoscaler`, the `metrics` configuration was a `v2beta2` feature and setting `metrics` would automatically make the resource use the v2beta2 API. So the only metric to base it on for the v1 version is `target_cpu_utilization_percentage`.

So this is available in `kubernetes_horizontal_pod_autoscaler_v2beta2` (see [documentation](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/horizontal_pod_autoscaler_v2beta2), can upgrade this resource to that if prefered?

Comment (gbmeuk) [reply k8s_pod_autoscaler.tf:23]: Yes, upgrade.

Comment (AlexBMes) [reply k8s_pod_autoscaler.tf]: Have updated it to use `kubernetes_horizontal_pod_autoscaler_v2`, and added the `metrics` block again.
This requires a minimum kubernetes version of v1.23 (which had an end of life date of 2023-02-28, so most clusters should be on a higher version).
Have updated documentation to address this

**Review by gbmeuk — APPROVED** [review]

**Pull request merged**